docs/internal-architecture.md

```diff
@@ -0,0 +1,53 @@
+## Internal architecture
+
+There are a few resources available to understand how the Collector works:
+
+### Startup Diagram
+```mermaid
+flowchart TD
+    A("`**command.NewCommand**`") -->|1| B("`**updateSettingsUsingFlags**`")
+    A --> |2| C("`**NewCollector**
+    Creates and returns a new instance of Collector`")
+    A --> |3| D("`**Collector.Run**
+    Starts the collector and blocks until it shuts down`")
+    D --> E("`**setupConfigurationComponents**`")
+    E -->  |1| F("`**getConfMap**`")
+    E ---> |2| G("`**Service.New**
+     Initializes telemetry, then initializes the pipelines`")
+    E --> |3| Q("`**Service.Start**
+    1. Start all extensions.
+    2. Notify extensions about Collector configuration
+    3. Start all pipelines.
+    4. Notify extensions that the pipeline is ready.
+    `")
+    Q --> R("`**Graph.StartAll**
+    Calls Start on each component in reverse topological order`")
+    G --> H("`**initExtensionsAndPipeline**
+     Creates extensions and then builds the pipeline graph`")
+    H --> I("`**Graph.Build**
+     Converts the settings to an internal graph representation`")
+    I --> J("`**createNodes**
+     Builds the node objects from pipeline configuration and adds to graph.  Also validates connectors`")
+    I --> K("`**createEdges**
+     Iterates through the pipelines and creates edges between components`")
+    I --> L("`**buildComponents**
+     Topological sort the graph, and create each component in reverse order`")
+    L --> M(Receiver Factory) & N(Processor Factory) & O(Exporter Factory) & P(Connector Factory)
+```
+### Important Files
```

Comment:
I feel like 'Important files' is a bit too vague. What are they important for? Different files are important to understand different parts of the codebase. 

The current list reads more like 'where to start to read the Collector codebase'. Maybe something like this is better?

```suggestion
### Where to start to read the code
```

I am not 100% convince by this title either, if you have better ideas let me know

Reply:
I originally had a brief description of what each file did - but the concern was this information belonged inside each file / at package level comments.

Reply:
Okay, missed that thread then, thanks for clarifying! (Feel free to drop).

Reply:
@mx-psi Yeah I don't have much of a better name - I think the paragraph under the heading is a good description of what the purpose of this section is.  Will try to tweak.